scipy/sparse/sputils.py

```diff
@@ -340,22 +340,15 @@ def is_pydata_spmatrix(m):
 ###############################################################################
 # Wrappers for NumPy types that are deprecated
 
-def matrix(*args, **kwargs):
-    with warnings.catch_warnings(record=True):
-        warnings.filterwarnings(
-            'ignore', '.*the matrix subclass is not the recommended way.*')
-        return np.matrix(*args, **kwargs)
+# Numpy versions of these functions raise deprecation warnings, the
+# ones below do not.
 
 
-def asmatrix(*args, **kwargs):
-    with warnings.catch_warnings(record=True):
-        warnings.filterwarnings(
-            'ignore', '.*the matrix subclass is not the recommended way.*')
-        return np.asmatrix(*args, **kwargs)
+def matrix(*args, **kwargs):
+    return np.array(*args, **kwargs).view(np.matrix)
 
 
-def bmat(*args, **kwargs):
```

Comment:
This is technically a compatibility break, right? Should we instead define our `bmat` here as an alias for `np.block`?

Reply:
I think these are not public functions?

Reply:
I'm sure that someone, somewhere is doing `from scipy.sparse.sputils import *` or equivalent. But I wasn't able to find any obvious instances with a cursory github search, so I think the risk is acceptable.

Reply:
Agreed `sputils` isn't public. If we have concerns about stuff like this, raising explicit deprecation warnings for importing from private modules like this may be a good idea.